### ISSUE TITLE
discuss git commit standards

### ISSUE THREAD

**Issue by TomMelt** [DataWaveProject/nonlocal_gwfluxes#16]
## Current problem

Currently viewing the `git log` I can see that there are a number of really long commit messages.

## 50/72 rule

A sensible guideline for commit message length is to follow the [50/72 rule](https://dev.to/noelworden/improving-your-commit-message-with-the-50-72-rule-3g79). I cannot find the original source of this rule, but you can read a lot of literature online about it.

**tldr**: title should be no more than 50 chars width and the body of the message should be no more than 72.

## semantic commit messages

As an additional bonus, I like to follow the [semantic/conventional commit](https://gist.github.com/joshbuchea/6f47e86d2510bce28f8e7f42ae84c716) message format.

**tldr**: for a commit that updates documentation e.g., the README. Use a commit message like follows:

## Example of the difference in readability

```
docs: added install instruction to README.md
```

This will make the git history significantly more readable on both GitHub and in the terminal 👍 

Example from our repo:
![Image](https://github.com/user-attachments/assets/47cdb9a7-20fc-42cc-a777-17c3227e41e9)

Example from my other repo:
![Image](https://github.com/user-attachments/assets/b02b2ef2-9cd3-4536-97b7-0a87f185a402)
